cypress/support/PersistentLogs.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.  
+ * Licensed under the MIT License.
+ */
+
+var testCaseRegistry = [];
+var indexRegistry = 0;
+var logEntries = '';
+
+// Override the original console.log() function
+const originalConsolLog = console.log;
+
+// This becomes the effective console.log() function.
+console.log = function (message) {
```

Comment:
Similarly all the vars can be `let`.  It generally goes like this. Use `const` everywhere by default and only change to `let` when you realize you need to re assign it. Shouldn't ever need to use `var` as it has all the javascript hoisting quirkiness and this makes it `undefined` in certain scenarios or available to used out of intended scope which aren't caught. `let` and `const` prevent these mistakes.

Reply:
I was not aware of these distinctions, so **thanks for pointing them out**. I did a little looking online and I came across this article which I agree with: [JavaScript ES6+: var, let, or const?](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75) so moving forward I will adopt this style.

But for the record, I disagree with the inventors of JavaScript including the `let` keyword in the language, I don't think it is necessary. However, since it is the norm, I am willing to go along to get along.

Reply:
Thanks for changing and thanks for replying on GitHub instead of email.

It's a bit ironic that the higher level point of the article you linked to is:
> keep things simple.

Variable types is only one small part of this. It reinforces the reason why I expressed concern over the complexity of test list manager, spec generator and such. They are not simple.  Since I can't offer alternatives it's best to go ahead but just keep this in mind.

Anyways at least the article helped explain the variable situation. Thanks